junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/DiscoverySelectors.java

```diff
@@ -592,6 +592,68 @@ public static MethodSelector selectMethod(Class<?> javaClass, String methodName,
 		return new MethodSelector(javaClass, methodName, parameterTypeNames.trim());
 	}
 
+	/**
+	 * Create a {@code MethodSelector} for the supplied class name, method name,
+	 * and parameter types.
+	 *
+	 * @param className the fully qualified name of the class in which the method
+	 * is declared, or a subclass thereof; never {@code null} or blank
+	 * @param methodName the name of the method to select; never {@code null} or blank
+	 * @param parameterTypes the formal parameter types of the method; never
+	 * {@code null} though potentially empty if the method does not declare parameters
+	 * @since 1.10
+	 * @see MethodSelector
+	 */
+	@API(status = EXPERIMENTAL, since = "1.10")
+	public static MethodSelector selectMethod(String className, String methodName, Class<?>... parameterTypes) {
+		return selectMethod((ClassLoader) null, className, methodName, parameterTypes);
+	}
+
+	/**
+	 * Create a {@code MethodSelector} for the supplied class name, method name,
+	 * parameter types, and class loader.
+	 *
+	 * @param classLoader the class loader to use to load the class, or {@code null}
+	 * to signal that the default {@code ClassLoader} should be used
+	 * @param className the fully qualified name of the class in which the method
+	 * is declared, or a subclass thereof; never {@code null} or blank
+	 * @param methodName the name of the method to select; never {@code null} or blank
+	 * @param parameterTypes the formal parameter types of the method; never
+	 * {@code null} though potentially empty if the method does not declare parameters
+	 * @since 1.10
+	 * @see MethodSelector
+	 */
+	@API(status = EXPERIMENTAL, since = "1.10")
+	public static MethodSelector selectMethod(ClassLoader classLoader, String className, String methodName,
+			Class<?>... parameterTypes) {
```

Comment:
I'm unsure about the overloads that take a `ClassLoader`, a `String` class name and a `Class<?>...` argument (such as here and below). Since the class is expected to be loaded by the supplied class loader it seems inconsistent. Did you add these overloads to supported mixed mode when indirectly called via the `@SelectMethod` annotation (in a future PR)?

Reply:
> Since the class is expected to be loaded by the supplied class loader it seems inconsistent. 

Yes, the `Class` will be loaded by the supplied `ClassLoader`.

But I don't understand what you think is inconsistent.

Can you please expound?

> Did you add these overloads to supported mixed mode when indirectly called via the `@SelectMethod` annotation (in a future PR)?

I added these overloads to align with other changes we have made in 1.10 to support the general "explicit ClassLoader for selectors" theme.

If we don't have these overloads, the user has no control over how the `Class` is loaded for the supplied `className`.

In other words, this aligns with the impetus for introducing `DiscoverySelectors.selectMethod(ClassLoader, String, String, String)` which has also been introduced in 1.10.

Having said that, I do concede that these particular overloads may not be used by many people. I added them for the sake of completeness, since I figured it wouldn't hurt. Although it does increase the number of `selectMethod(...)` overloads. 

If you feel that we should not introduce these overloads, I can remove them, and we can add them later if requested by the community.

Reply:
Thanks for the clarification.

I'll remove those variants.

If we ever decide we need them, they can be recovered from the commit history in this PR.

Reply:
Done in 74e78639ccc83d9bcaac600fb5563c9d5054172e

Reply:
> I meant that the test class will be loaded by the supplied class loader while the parameter types won't. If we don't have a use case for these right now I'd therefore omit them.

Sorry for being late to this discussion... this is a little bit complicated, but for posterity/future reference:

While it *can* be a problem if the parameters are loaded by a different classloader, it not *necessarily* a problem. In fact, in a multiple-classloader environment like OSGi, there are common circumstances where the parameter types *must* be loaded by a different classloader. Consider, for example, if you have a test class in a custom bundle - it will be loaded by the bundle's classloader. However, if it has a test method which takes a Java primitive as a parameter like `java.lang.String`, `java.lang.int` or `java.lang.Integer`, then the parameter will be loaded by the system classloader.

The more important question rather than "is it a different classloader" is "are the classes visible to the test class' classloader". The answer to this question in general depends the specific classloader hierarchy/environment, but it is certainly possible with the `(ClassLoader, String, String, Class<?>...)` variant to end up passing `Class` instances as parameters that are not visible to the test class' classloader and hence won't load. The same is also true of the `(ClassLoader, String, String, String...)` variant though, as you could pass in FQN of parameter types that are not visible to the `ClassLoader` you've provided.

Reply:
Thanks for chiming in, @kriegfrj.

> The answer to this question in general depends the specific classloader hierarchy/environment

I think that pretty much sums it up: _it depends_. 😉 

In any case, we'll go with what we have in `main` for the time being, and we can revisit this later if necessary.